yaml/build.go

```diff
@@ -63,22 +65,31 @@ func handleBuildArgs(value interface{}) (map[string]string, error) {
 	case map[interface{}]interface{}:
 		return handleBuildArgMap(v)
 	case []interface{}:
-		args = map[string]string{}
-		for _, elt := range v {
-			uniqArgs, err := handleBuildArgs(elt)
-			if err != nil {
-				return args, nil
-			}
-			for mapKey, mapValue := range uniqArgs {
-				args[mapKey] = mapValue
-			}
-		}
-		return args, nil
+		return handleBuildArgSlice(v)
 	default:
 		return args, fmt.Errorf("Failed to unmarshal Build args: %#v", value)
 	}
 }
 
+func handleBuildArgSlice(s []interface{}) (map[string]string, error) {
+	var args = map[string]string{}
+	for _, arg := range s {
+		// check if a value is provided
+		switch v := strings.SplitN(arg.(string), "=", 2); len(v) {
+		case 1:
+			// we don't use EnvironmentLookup here as docker does not use values in the env_file or environment config stanza for this
+			envValue := os.Getenv(v[0])
```

Comment:
So, docker-compose itself doesn't read build "Args" values from the environments stanza or the env_file entry.   Is that something that you would want to support for libcompose?   Env values and build Args are handled very differently in both docker itself and docker-compose.   Having libcompose populate build Args from Env values that are specified in environments or env_file would be a significant deviation from the standard behavior...